nstar/src/main.rs

```diff
@@ -99,37 +114,39 @@ async fn main() -> Result<()> {
                     } else {
                         break;
                     }
-                }
-                input = input.next() => {
-                    let input: &str = if input.is_some() {
-                        input.as_deref().unwrap()
-                    } else {
-                        break 'outer;
+                },
+                input = terminal.readline() => {
+                    let input: String = match input {
+                        Ok(line) => line,
+                        Err(ReadlineError::Interrupted) | Err(ReadlineError::Eof) => return Ok(()),
+                        Err(e) => {
+                            writeln!(terminal, "{}", e.to_string().red())?;
+                            break
+                        },
                     };
-                    match parse_prompt(input) {
+
+                    match prompt.parse(&input) {
                         Ok(PromptCommand::Prompt(cmd)) => match cmd {
-                            Prompt::Help => print_help(&mut terminal)?,
-                            Prompt::Quit => break 'outer,
+                            Prompt::Help => prompt.print_help(&mut terminal)?,
+                            Prompt::Quit => return Ok(()),
                         },
                         Ok(PromptCommand::Northstar(cmd)) => {
                             match send_request(&mut client, cmd).await {
                                 Ok(response) => print_response(&mut terminal, response, opt.json).await?,
                                 // break the loop and try to reconnect
                                 Err(api::client::Error::Stopped) => break,
-                                Err(e) => writeln!(&mut terminal, "{}: {}", "client error".red(), e)?,
+                                Err(e) => writeln!(terminal, "{}: {}", "client error".red(), e)?,
                             };
                         },
-                        Err(e) => writeln!(&mut terminal, "{}", e.message)?
+                        Err(e) => writeln!(terminal, "{}", e.message)?
                     };
+                    terminal.flush()?;
                 }
             }
         }
 
-        writeln!(terminal, "Disconnected")?;
-        time::sleep(time::Duration::from_secs(1)).await;
-
         // Try to reconnect
-        client = create_client(&opt.host).await?;
+        writeln!(terminal, "Disconnected, trying to reconnect")?;
+        client = try_connect(&mut terminal, &url).await?;
```

Comment:
while this is already very good, we could also do a less verbose reconnect by catching failed connects and just adding a `.` to the output

Reply:
Oh yes, I like this idea :)